backend/src/v2v3/dex/data.ts

```diff
@@ -0,0 +1,182 @@
+import { bqClient } from "../../bigQuery";
+import { redisClient, getOrThrow } from "../../redisSetup";
+import { getErrorMessage, getPrice } from "./utils";
+
+const bigQueryEndpointBase = "crypto-stellar.crypto_stellar_2";
+
+async function fetchBigQueryData(query: string) {
+  try {
+    const [job] = await bqClient.createQueryJob(query);
+    console.log("running bq query:", query);
+    const [result] = await job.getQueryResults();
+    return result;
+  } catch (error: unknown) {
+    const message = getErrorMessage(error);
+    console.error("QUERY ERROR: ", message);
+    return null;
+  }
+}
+
+async function fetchMemoizedData(key: string, query: string) {
+  try {
+    const cachedData = await getOrThrow(redisClient, key);
+    const output = JSON.parse(cachedData);
+    return output;
+  } catch (error: unknown) {
+    const message = getErrorMessage(error);
+    if (message == "redis key not found") {
+      const output = await fetchBigQueryData(query);
+      await redisClient.set(key, JSON.stringify(output));
+      return output;
+    }
+    console.error("ERROR: ", message);
+  }
+}
+
+export async function getPaymentsData() {
+  const query = `
+    SELECT count(*) as data 
+    FROM ${bigQueryEndpointBase}.enriched_history_operations 
+    WHERE type in (1, 2, 13) 
+    AND closed_at >= TIMESTAMP_ADD(CURRENT_TIMESTAMP(), INTERVAL -24 HOUR) 
+    AND successful is true
+  `;
+  const payments = await fetchMemoizedData("dex-payments24h", query);
```

Comment:
Does the dashboard have a similar label for `dex-payments-24h`? Technically non path-payment operations of type 1 do not go through the DEX to execute.

Reply:
oh ya  I guess this technically isn't "dex" data, it should just be grabbing the last 24h payments.

@Selhar we could probably change this label to just `payments24h`?

@sydneynotthecity (not sure if you have access to these designs), but these are the metrics were looking to grab:

![Screen Shot 2022-05-26 at 11 19 01 AM](https://user-images.githubusercontent.com/30449853/170551142-92b47d56-ec52-48ed-ac65-b7fb73aa534a.png)

Reply:
Excellent, just making sure you're representing just `payments` instead of `dex-payments`. Looks good!

Reply:
Fixed the label and looking into the updated query (hadn't seen the comment before). Pushing new commits shortly.